packages/ios_platform_images/lib/ios_platform_images.dart

```diff
@@ -138,6 +138,74 @@ class IosPlatformImages {
     return _FutureMemoryImage(bytesCompleter.future, scaleCompleter.future);
   }
 
+  /// Loads an image from the system.  The equivalent would be:
+  /// `[UIImage systemImageNamed:name]`.
+  ///
+  /// Throws an exception if the image can't be found.
+  ///
+  /// [colors] takes a list of colors to be applied to the corresponding layer
+  /// of the icon, assuming that layer exists. An icon can have up to 3 layers,
+  /// if a layer color is not specified it will take the color of the most recent
+  /// valid index.
+  ///
+  /// For more information see [https://developer.apple.com/documentation/uikit/uiimagesymbolconfiguration/3810054-configurationwithpalettecolors?language=objc]
+  ///
+  /// [preferMulticolor] overrides [colors] and asks iOS to provide its preset
+  /// multicolor varient of the symbol. Depending on the symbol, these colors
+  /// *may not be mutable*. To find out, use the SF Symbols app found at
+  /// [https://developer.apple.com/sf-symbols/]
+  ///
+  /// Multi-layer and multi-color symbols only work on iOS 15.0 and above.
+  /// **This method requires at least iOS 13.0**
+  ///
+  /// See [https://developer.apple.com/documentation/uikit/uiimage/configuring_and_displaying_symbol_images_in_your_ui?language=objc]
+  static ImageProvider loadSystemImage(
+    String name,
+    double pointSize, {
+    List<Color> colors = const <Color>[],
+    bool preferMulticolor = false,
+    int weightIndex = 1,
```

Comment:
My philosophy was to leave any widget-specific stuff to the eventual implementation in a separate `sfsymbols` package, which would have access to the BuildContext and the theme information. I imagined this as more of a bare-bones method, so I came up with some basic defaults.

That's just what made sense to me as I was writing it, but I could see an argument for getting the build context here.

Reply:
> My philosophy was to leave any widget-specific stuff to the eventual implementation in a separate `sfsymbols` package, which would have access to the BuildContext and the theme information. I imagined this as more of a bare-bones method, so I came up with some basic defaults.
> 
> That's just what made sense to me as I was writing it, but I could see an argument for getting the build context here.

If someone didn't use your eventual-sfsymbols package, is there any advantage to making the user think in terms of `weightIndex` instead of `FontWeight`, or similar Flutter idioms?  It's importing Flutter (it's not a generic dart package) so may as well use the higher level abstractions Flutter provides.

Reply:
Okay, this is a valid point I agree with; I've been trying to implement this across the board, but I've come to an issue regarding the size. From my understanding Flutter `iconSize` is in logical pixels but the SF Symbols take a font point size. 

I've spent the weekend trying to figure out how to convert from pixels to font size, but I couldn't find a reliable method. How would you recommend approaching this?

Reply:
Maybe pass in the build context and fetch `MediaQuery.of(context).devicePixelRatio`?

@goderbauer @Piinks this isn't in the `flutter` package but this feels framework-y, can you make a suggestion for how to make this API more idiomatic looking?

Reply:
For a low-level image loading API this API seems fine. I don't think I would pass in the build context here to retain the flexibility of just using this in the render layer directly (where you don't have a build context). However, it would be nice to have a higher-level wrapper API (maybe a widget) that does read some of these things from e.g. a theme.

Reply:
> For a low-level image loading API this API seems fine. I don't think I would pass in the build context here to retain the flexibility of just using this in the render layer directly (where you don't have a build context). However, it would be nice to have a higher-level wrapper API (maybe a widget) that does read some of these things from e.g. a theme.

Fair enough, that higher-level API will be my next PR after this one in a separate package; the discussion is on https://flutter.dev/go/sfsymbols. Do you have any thoughts on converting logical pixels to point size? 

I'm not sure if GitHub pings you anyway, but I'm going to add a @goderbauer for good measure.

Reply:
What is "point size"? Is that physical pixels? If it is that you'd use the devicepixelratio.

Reply:
Let me take a step back first; the ultimate issue is that a size 100 Material icon is not the same as a size 100 SFSymbol, which is unintuitive for anyone using the API.

<img src="https://user-images.githubusercontent.com/13630061/156238177-8ab6f034-c762-4f28-a31f-779d025f117d.png" width="75">

The tl;dr is that it looks like iOS has a point size (pixels scaled up for 2x or 3x displays similar to logical pixels) and a font point size that is possibly defined as 1/72 of an inch.

I've done a bunch of digging and found that the [`UIImageSymbolConfiguration`](https://developer.apple.com/documentation/uikit/uiimagesymbolconfiguration/3294241-configurationwithpointsize?language=objc) takes a system font point size. Apple docs aren't entirely clear on what this means, but [some posts suggest](https://stackoverflow.com/a/1059323/5565806) it takes the definition of a font point as 1/72 of an inch. There was also a [discussion](https://stackoverflow.com/questions/11776767/what-exactly-is-uifonts-point-size) about the correlation between font and pixel sizes that may be useful.

I've also discussed this in the Flutter Discord in [this thread](https://canary.discord.com/channels/608014603317936148/946974368385351681).

Reply:
Friendly ping to @goderbauer; I'm pretty sure sizing should be one of if not the last tweak before this can land.

Reply:
The "this thread" link is not going anywhere for me... So, I don't know what has been discussed there...

Ideally, the size you pass in here should be the same unit you use everywhere else in your Flutter app. So if I specify 100 here, that should be the same as a SizedBox with 100 as a dimension. Everything else will be very confusing. I don't know enough about iOS, though, to say how to convert from Flutter's unit to what iOS expects...